src/compile/mark/valueref.ts

```diff
@@ -206,7 +208,14 @@ export function tooltipForChannelDefs(channelDefs: FieldDef<string>[], config: C
   const keyValues: string[] = [];
   const usedKey = {};
   for (const fieldDef of channelDefs) {
-    const key = getFirstDefined(fieldDef.title, title(fieldDef, config));
+    let key;
+    if (isPositionFieldDef(fieldDef) && fieldDef.axis && fieldDef.axis.title) {
```

Comment:
Maybe better include this logic in the `title()` function that you call here, so it's reusable in many other places. 

Also, check if it is FacetFieldDef and access `header.title`.

This is a good opportunity to clean this up.
(Try find all references of `title?:` in `TitleMixins` and all references of the `title()` method.)